commcare_connect/opportunity/models.py

```diff
@@ -601,3 +601,9 @@ class CatchmentArea(models.Model):
     opportunity_access = models.ForeignKey(OpportunityAccess, null=True, on_delete=models.DO_NOTHING)
     active = models.BooleanField(default=True)
     name = models.CharField(max_length=255)
+    site_code = models.SlugField(max_length=255, unique=True, blank=True)
+
+    def save(self, *args, **kwargs):
```

Comment:
I don't think we want this logic in the save function. I think the import is enough, and that it needs to be a bit different (as I mention there). Basically, use the provided code unless it doesn't exist in which case sluggify the name

Reply:
It is probably worth requiring a site code at this point, actually. Just raise an upload error if it is not included, then we don't have to guess.

Reply:
@calellowitz Added the requested changes.